pkg/apis/kubermatic/v1/datacenter.go

```diff
@@ -90,6 +90,66 @@ func IsProviderSupported(name string) bool {
 	return false
 }
 
+// +kubebuilder:validation:Enum="";Healthy;Unhealthy;Invalid;Terminating;Paused
+
+type SeedPhase string
+
+// These are the valid phases of a seed.
+const (
+	// SeedHealthyPhase means the seed is reachable and was successfully reconciled.
+	SeedHealthyPhase SeedPhase = "Healthy"
+
+	// SeedUnhealthyPhase means the KKP resources on the seed cluster could not be
+	// successfully reconciled.
+	SeedUnhealthyPhase SeedPhase = "Unhealthy"
+
+	// SeedInvalidPhase means the seed kubeconfig is defunct.
+	SeedInvalidPhase SeedPhase = "Invalid"
+
+	// SeedTerminatingPhase means the seed is currently being deleted.
+	SeedTerminatingPhase SeedPhase = "Terminating"
+
+	// SeedPausedPhase means the seed is not being reconciled because if SkipReconciling
+	// annotation is set.
+	SeedPausedPhase SeedPhase = "Paused"
+)
+
+// +kubebuilder:validation:Enum="";SeedConditionResourcesReconciled;SeedConditionValidKubeconfig;
+
+// SeedConditionType is used to indicate the type of a seed condition. For all condition
+// types, the `true` value must indicate success. All condition types must be registered
+// within the `AllSeedConditionTypes` variable.
+type SeedConditionType string
+
+const (
+	// SeedConditionValidKubeconfig indicates that the configured kubeconfig for the seed is valid.
+	// The seed-sync controller manages this condition.
+	SeedConditionValidKubeconfig SeedConditionType = "ValidKubeconfig"
+	// SeedConditionResourcesReconciled indicates that the KKP operator has finished setting up the
+	// resources inside the seed cluster.
+	SeedConditionResourcesReconciled SeedConditionType = "ResourcesReconciled"
+)
+
+var AllSeedConditionTypes = []SeedConditionType{
+	SeedConditionResourcesReconciled,
+}
+
+type SeedCondition struct {
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time we got an update on a given condition.
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime"`
```

Comment:
Isn't this usually called `LastProbeTime`?

Reply:
I copied this from the ClusterConditions, which also use LastHeartbeatTime. I think it makes sense to be consistent?

Reply:
`LastProbeTime` is in Pods for when the last probe ran, but it's a Pod specific thing I believe. Nodes also have `LastHeartbeatTime`.